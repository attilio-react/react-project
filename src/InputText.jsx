import React from 'react';

class InputText extends React.PureComponent {
  render () {
    return <input type="text" value={this.props.inputValue}></input>;
  }
}


export {InputText}


