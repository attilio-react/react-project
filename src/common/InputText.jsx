import React from 'react';

class InputText extends React.PureComponent {
  render () {
    return <input type="text" onChange={this.props.onChange} value={this.props.inputValue}></input>;
  }
}


export {InputText}


