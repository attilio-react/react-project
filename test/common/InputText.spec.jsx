import * as React from 'react';
import {mount} from 'enzyme';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {InputText} from '|common/InputText.jsx';

enzyme.configure({adapter: new Adapter()});

describe('InputText rendering', () => {
  it('Should render', () => {
    const component = renderer.create(<InputText value='hello' />),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('InputText behaviour', () => {
  it('Should call callback when changed', () => {
    const onChange = jest.fn(),
      input = mount(<InputText value='hello' onChange={onChange} />),
      nativeInput = input.find('input').first();
    nativeInput.simulate('change');
    expect(onChange).toBeCalled()
  });
});

