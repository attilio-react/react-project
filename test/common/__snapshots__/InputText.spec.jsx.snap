// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText rendering Should render 1`] = `
<input
  type="text"
/>
`;
